platform/brickpi3-motor-and-sensor.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+"""
+The Brickpi3 doesn't support auto-detecting motors and sensors. To use devices
+connected to the LEGO ports, you must specify what type of device it is.
+"""
+
+from time import sleep
+from ev3dev2 import list_devices
+from ev3dev2.port import LegoPort
+from ev3dev2.motor import OUTPUT_A, LargeMotor, SpeedPercent
+from ev3dev2.sensor import INPUT_1
+from ev3dev2.sensor.lego import UltrasonicSensor
+
+p1 = LegoPort(INPUT_1)
+# http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-stretch/brickpi3.html#brickpi3-in-port-modes
+p1.mode = 'ev3-uart'
+# http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-stretch/sensors.html#supported-sensors
+p1.set_device = 'lego-ev3-us'
+
+pA = LegoPort(OUTPUT_A)
+# http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-stretch/brickpi3.html#brickpi3-out-port-modes
+pA.mode = 'tacho-motor'
+# http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-stretch/motors.html#supported-motors
+pA.set_device = 'lego-ev3-l-motor'
```

Comment:
BrickPi3 output ports are pre-configured for NXT motors, so the port manipulation here isn't strictly necessary.

Reply:
But maybe trying to explain that is more confusing 😄

Reply:
My inclination is to opt for the "one size fits all" solution if there's ever a case where another motor type is preferred. If using other types of motors is rare (maybe it is?) then it might be worth just leaving this commented out entirely with a note about it.

Reply:
It's worth noting the touch sensor is in a similar position (from my reading of the docs, anyway). I think the "device" is already set once you configure the mode. Not sure if that factors in here.

Reply:
So it looks like `set_device` on output ports is not supported.

```
Traceback (most recent call last):
  File "/home/robot/brickpi3/main.py", line 25, in <module>
    pA.set_device = 'lego-ev3-l-motor'
  File "/usr/lib/python3/dist-packages/ev3dev2/port.py", line 140, in set_device
    self._set_device = self.set_attr_string(self._set_device, 'set_device', value)
  File "/usr/lib/python3/dist-packages/ev3dev2/__init__.py", line 322, in set_attr_string
    return self._set_attribute(attribute, name, value)
  File "/usr/lib/python3/dist-packages/ev3dev2/__init__.py", line 264, in _set_attribute
    self._raise_friendly_access_error(ex, name, value)
  File "/usr/lib/python3/dist-packages/ev3dev2/__init__.py", line 288, in _raise_friendly_access_error
    raise driver_error
  File "/usr/lib/python3/dist-packages/ev3dev2/__init__.py", line 261, in _set_attribute
    attribute.write(value)
OSError: [Errno 95] Operation not supported
```